src/main.rs

```diff
@@ -1,3 +1,95 @@
+mod constants;
+mod notifier;
+
+use crate::constants::{
+    LINK, MAX_NUMBER_OF_TRIES, SECS_BETWEEN_REQUESTS, SECS_TO_SLEEP_AFTER_REQUEST_ERROR, SELECTOR,
+};
+use crate::notifier::TelegramNotifier;
+use scraper::{Html, Selector};
+use std::thread::sleep;
+use std::time::Duration;
+use anyhow::bail;
+use tracing_appender::rolling;
+use tracing_subscriber::fmt::writer::MakeWriterExt;
+
+type NumberOfAds = usize;
+
 fn main() {
-    println!("Hello, world!");
+    dotenv::dotenv().unwrap();
+
+    let log_file = rolling::daily("./logs", "app");
+    tracing_subscriber::fmt()
+        .json()
+        .with_writer(log_file.and(std::io::stdout))
+        .init();
+
+    tracing::info!("starting app");
+
+    let tg_notifier = TelegramNotifier::new(
+        std::env::var("TG_BOT_TOKEN").unwrap(),
+        std::env::var("TG_CHAT_ID").unwrap(),
+    );
+
+    tg_notifier.send_message("test message").unwrap();
+
+    let time_between_requests = Duration::from_secs(SECS_BETWEEN_REQUESTS);
+    let mut number_of_ads = get_number_of_ads(&tg_notifier);
+    sleep(time_between_requests);
+
+    loop {
+        let new_number_of_ads = get_number_of_ads(&tg_notifier);
+
+        if new_number_of_ads > number_of_ads {
```

Comment:
You can future proof the logic of checking updates. So that when you decided to update it, it won't force you to change main code.

It could be isolated as a separate structure, which would have method kinda ".check_updates()", that would return a standardized answer

Reply:
yeap, extracting the checking logic would be great!